chromadb/test/property/test_filtering.py

```diff
@@ -100,6 +100,13 @@ def _filter_where_doc_clause(clause: WhereDocument, doc: Document) -> bool:
             expr = expr.replace("%", ".").replace("_", ".")
             return re.search(expr, doc) is not None
         return expr in doc
+    elif key == "$not_contains":
+        # SQLite FTS handles % and _ as word boundaries that are ignored so we need to
+        # treat them as wildcards
+        if "%" in expr or "_" in expr:
+            expr = expr.replace("%", ".").replace("_", ".")
+            return doc and re.search(expr, doc) is None
+        return doc and expr not in doc
```

Comment:
Nit: I don't love that we check `doc and ...` for this if-branch but not the above branch.

Could we instead do one of:
- Do an `if not doc` check above so both branches benefit from it
- Also do the `doc and ...` check in the `key == "$contains"` branch
- Not do the check here (not possible if the typechecker is complaining)?

Reply:
I will add the check above the if-branch.
this check exists because SQLite 'not like' does not return rows for empty or NULL strings. In this case, both "contains" and "not_contains" will not return NULL or empty strings. We can also argue that "not_contains" should return all NULL and empty strings.